src/components/popup.js

```diff
@@ -179,7 +179,12 @@ function Popup(props) {
       ref={containerRef}
     >
       <div key="tip" className="mapboxgl-popup-tip" style={{borderWidth: tipSize}} />
-      <div key="content" ref={contentRef} className="mapboxgl-popup-content" onClick={onReactClick}>
+      <div
+        key="content"
+        ref={contentRef}
+        className="mapboxgl-popup-content"
+        onClick={e => onReactClick(e, {props, context})}
```

Comment:
`onReactClick` can be `null`. You can do

```js
const thisRef = useMapControl({...props, onClick});
const {context, containerRef} = thisRef;

...

const onReactClick = useCallback(e => !context.eventManager && onClick(e, thisRef), []);

Reply:
🧹 thank you